pkg/providers/tinkerbell/create.go

```diff
@@ -80,15 +80,20 @@ func (p *Provider) PostBootstrapSetup(ctx context.Context, clusterConfig *v1alph
 func (p *Provider) PostWorkloadInit(ctx context.Context, cluster *types.Cluster, clusterSpec *cluster.Spec) error {
 	logger.V(4).Info("Installing Tinkerbell stack on workload cluster")
 
+	if p.datacenterConfig.Spec.SkipLBSetup {
+		logger.Info("Warning: Skipping load balancer deployment. Please install and configure a load balancer once the cluster is created.")
+	}
+
 	err := p.stackInstaller.Install(
 		ctx,
 		clusterSpec.VersionsBundle.Tinkerbell,
 		p.templateBuilder.datacenterSpec.TinkerbellIP,
 		cluster.KubeconfigFile,
 		p.datacenterConfig.Spec.HookImagesURLPath,
 		stack.WithBootsOnKubernetes(),
-		stack.WithHostPortEnabled(false),
-		stack.WithLoadBalancer(),
+		stack.WithHostPortEnabled(false), // disable host port on workload cluster
+		stack.WithEnvoyEnabled(true), // use envoy on workload cluster
```

Comment:
so envoy installation is not optional, only kube-vip is? that should be okay for now... not sure if for single node resource conservation concerns, at some point we will need to make envoy optional

Reply:
I would recommend keeping envoy even for single node. It has a really small footprint and prevents us from having to expose two load balancer services on the same IP, which some LBs might not support